src/main/java/net/minecraftforge/snowblower/Generator.java

```diff
@@ -133,7 +149,58 @@ public Generator setup(String branchName, @Nullable String remoteName, Config cf
         return this;
     }
 
+    private void setupRemote(@Nullable URL remoteUrl) throws GitAPIException {
+        if (remoteUrl == null)
+            return;
+
+        URIish remoteFakeUri = new URIish(remoteUrl);
+        String foundRemote = null;
+
+        Set<String> remoteNames = new HashSet<>();
+
+        for (RemoteConfig remoteConfig : this.git.remoteList().call()) {
+            String currRemoteName = remoteConfig.getName();
+            remoteNames.add(currRemoteName);
+
+            if (foundRemote != null)
+                continue;
+
+            for (URIish fakeUri : remoteConfig.getURIs()) {
+                if (fakeUri.equals(remoteFakeUri)) {
+                    foundRemote = currRemoteName;
+                    break;
+                }
+            }
+        }
+
+        if (foundRemote == null) {
+            int i = 0;
+            String attemptRemoteName = "origin";
+            while (remoteNames.contains(attemptRemoteName)) {
+                i++;
+                attemptRemoteName = "origin" + i;
+            }
+
+            this.git.remoteAdd().setName(attemptRemoteName).setUri(remoteFakeUri).call();
+            this.removeRemote = true;
+        }
+
+        this.remoteName = foundRemote;
+
+        this.git.fetch();
```

Comment:
The command is never called and for big remotes it gives the false impression of it hanging, so please log it.
```suggestion
        this.git.fetch().setRemote(remoteName).setProgressMonitor(new TextProgressMonitor(new OutputStreamWriter(System.out))).call();
```

Reply:
damn